websocket/server.go

```diff
@@ -229,8 +242,7 @@ func (server *Server) Run(ctx context.Context) {
 
 // actuateRegisterClient is what the Server uses to actually register a Client.
 // NOTE: This may be called on a goroutine, as it acts in reaction to data sent into a channel.
-// DO NOT USE THIS METHOD OUTSIDE THIS FILE, PLEASE.
```

Comment:
Instead of this comment, make this be a method of `*server` , so it's clear it should only be used as an implementation of it. Also, since `*server` does not leave this file implementation and other modules will only interact with the `Server interface` and this method is unexported, there is very little risk of it being misused.

Reply:
Good idea. But still they are in the same package. So, at some it could be refactored and broken into separate packages so they really can’t see each other’s “private” properties, and this ^^